xwiki-platform-core/xwiki-platform-notifications/xwiki-platform-notifications-ui/src/main/resources/XWiki/Notifications/Code/NotificationsFiltersPreferencesMacro.xml

```diff
@@ -637,14 +658,15 @@ require(['jquery', 'AddNotificationFilterPreferenceLivetable', 'bootstrap', 'boo
       }).done(function() {
         notif.hide();
         new XWiki.widgets.Notification("$escapetool.javascript($services.localization.render('notifications.filters.preferences.setEnabled.done'))", 'done');
+        updateFetchedRow(data, state);
       }).fail(function() {
         notif.hide();
         new XWiki.widgets.Notification("$escapetool.javascript($services.localization.render('notifications.filters.preferences.setEnabled.error'))", 'error');
       });
     });
     // Enable / Disable action for toggleable filter preferences
     $(data.row).find('.toggleableFilterPreferenceCheckbox').each(function() {
-      new ToggleableFilterPreference(this, userURL);
+      new ToggleableFilterPreference(this, userURL, data, updateFetchedRow);
```

Comment:
It seems you pass ``data`` just to get it back in ``updateFetchedRow``, so you could write instead:

    new ToggleableFilterPreference(this, userURL, function(state) {
        updateFetchedRow(data, state);
    });

You could also trigger an event when the state is saved (both for ``notificationFilterPreferenceCheckbox`` and for ``toggleableFilterPreferenceCheckbox``) and update the fetched row in an event listener.

Reply:
Thanks I'll refactor the code. I first wanted to make sure that it was how we wanted to fix the issue.